code/modules/mob/living/carbon/status_procs.dm

```diff
@@ -63,6 +63,43 @@
 		clear_fullscreen("high")
 		clear_alert("high")
 
+/mob/living/carbon/adjust_disgust(amount)
+	var/old_disgust = disgust
+	if(amount>0)
+		disgust = min(disgust+amount, DISGUST_LEVEL_MAXEDOUT)
+		switch(disgust)
```

Comment:
if it was just a stomach thing I would've called it food poisoning or whatever, I want to keep it so you could possibly be disgusted by other stuff besides your stomach in the future.